ffi-convert/src/conversions.rs

```diff
@@ -107,22 +131,48 @@ pub struct UnexpectedNullPointerError;
 
 /// Trait representing the creation of a raw pointer from a struct and the recovery of said pointer.
 ///
-/// The `from_raw_pointer` function should be used only on pointers obtained thought the
+/// The `from_raw_pointer` function should be used only on pointers obtained through the
 /// `into_raw_pointer` method (and is thus unsafe as we don't have any way to get insurance of that
 /// from the compiler).
 ///
 /// The `from_raw_pointer` effectively takes back ownership of the pointer. If you didn't create the
 /// pointer yourself, please use the `as_ref` method on the raw pointer to borrow it
-///
-/// A generic implementation of this trait exist for every struct, it will use a `Box` to create the
-/// pointer. There is also a special implementation available in order to create a
-/// `*const libc::c_char` from a CString.
 pub trait RawPointerConverter<T>: Sized {
     fn into_raw_pointer(self) -> *const T;
+    fn into_raw_pointer_mut(self) -> *mut T;
     unsafe fn from_raw_pointer(input: *const T) -> Result<Self, UnexpectedNullPointerError>;
+    unsafe fn from_raw_pointer_mut(input: *mut T) -> Result<Self, UnexpectedNullPointerError>;
 
     unsafe fn drop_raw_pointer(input: *const T) -> Result<(), UnexpectedNullPointerError> {
-        T::from_raw_pointer(input).map(|_| ())
+        Self::from_raw_pointer(input).map(|_| ())
+    }
+
+    unsafe fn drop_raw_pointer_mut(input: *mut T) -> Result<(), UnexpectedNullPointerError> {
+        Self::from_raw_pointer_mut(input).map(|_| ())
+    }
+}
+
+pub fn convert_into_raw_pointer<T>(pointee: T) -> *const T {
```

Comment:
as this is public we should document it (or put a #[doc(hidden)] on it)

Reply:
Correct ! I will hide it from the documentation though.